src/Raven.Server/Documents/ETL/Relational/RelationalDocumentTransformer.cs

```diff
@@ -0,0 +1,243 @@
+
+    private EtlStatsScope _stats;
+
+    public RelationalDocumentTransformer(Transformation transformation, DocumentDatabase database, DocumentsOperationContext context, TRelationalEtlConfiguration config, PatchRequestType patchRequestType)
+        : base(database, context, new PatchRequest(transformation.Script, patchRequestType), null)
+    {
+        _transformation = transformation;
+        _config = config;
+
+        var destinationTables = transformation.GetCollectionsFromScript();
+
+        LoadToDestinations = destinationTables;
+
+        _tables = new Dictionary<string, RelationalTableWithRecords>(destinationTables.Length, StringComparer.OrdinalIgnoreCase);
+        _tablesForScript = new List<RelationalTableWithRecords>(destinationTables.Length);
+
+
+        _etlTablesWithRecords = GetEtlTables();
+        // ReSharper disable once ForCanBeConvertedToForeach
+        for (var i = 0; i < _etlTablesWithRecords.Count; i++)
+        {
+            var table = _etlTablesWithRecords[i];
+
+            if (destinationTables.Contains(table.TableName, StringComparer.OrdinalIgnoreCase))
+                _tablesForScript.Add(table);
+        }
+
+        if (_transformation.IsLoadingAttachments)
+           _loadedAttachments = new Dictionary<string, Queue<Attachment>>(StringComparer.OrdinalIgnoreCase);
+    }
+
+    protected abstract List<RelationalTableWithRecords> GetEtlTables();
+
+    public override void Initialize(bool debugMode)
+    {
+        base.Initialize(debugMode); // todo: explain varchar vs nvarchar?
+        
+        DocumentScript.ScriptEngine.SetValue("varchar",
+            new ClrFunction(DocumentScript.ScriptEngine, "varchar", (value, values) => ToVarcharTranslator(VarcharFunctionCall.AnsiStringType, values)));
+
+        DocumentScript.ScriptEngine.SetValue("nvarchar",
+            new ClrFunction(DocumentScript.ScriptEngine, "nvarchar", (value, values) => ToVarcharTranslator(VarcharFunctionCall.StringType, values)));
+    }
+
+    protected override string[] LoadToDestinations { get; }
+
+    protected override void LoadToFunction(string tableName, ScriptRunnerResult cols)
+    {
+        if (tableName == null)
+            ThrowLoadParameterIsMandatory(nameof(tableName));
+
+        var result = cols.TranslateToObject(Context);
+        var columns = new List<RelationalColumn>(result.Count);
+        var prop = new BlittableJsonReaderObject.PropertyDetails();
+
+        for (var i = 0; i < result.Count; i++)
+        {
+            result.GetPropertyByIndex(i, ref prop);
+
+            var relationalColumn = new RelationalColumn { Id = prop.Name, Type = prop.Token, Value = prop.Value };
+
+            if (_transformation.IsLoadingAttachments && 
+                prop.Token == BlittableJsonToken.String && IsLoadAttachment(prop.Value as LazyStringValue, out var attachmentName))
+            {
+                var attachment = _loadedAttachments[attachmentName].Dequeue();
```

Comment:
Are we going to support attachments in Snowflake ETL as well?

Reply:
There is BINARY type in Snowflake, which will work for attachments under 8MB.

Reply:
https://docs.snowflake.com/en/sql-reference/data-types-text#binary

Reply:
Great, so we can support that.

Reply:
done